### DISCUSSION TITLE
Server : different access level for different client

### DISCUSSION THREAD

**Discussion by Thomangin** [FreeOpcUa/opcua-asyncio#832] [Q&A] [answered]
Hello all,

I have a little server running, giving access to a few variables. Two different clients may connect to the server at the same time : what I would like to do is giving one client the access to read and write every variable, and the other could only read.

First I thought about doing that by "creating" two different users, but as far as I understood this is not implemented yet (?). If so, is there any other way I could achieve this goal? 

Looking at the doc, I see that there is some kind of user management in the internal server class (which is a concept I still not fully understand) , but there is not so much description, nor example, and I'll admit I'm a bit lost. 

Thank you in advance.

**Comment by AndreasHeine** [comment, marked as answer]
yes Rolepermissions are not implemented just basic admin/user but it should do what you like to have.
you need to implement a usermanager like in https://github.com/FreeOpcUa/opcua-asyncio/blob/04a85e2c5bd16c353e445ad8b691bd3868a69954/asyncua/server/user_managers.py#L11
pass it to:
https://github.com/FreeOpcUa/opcua-asyncio/blob/04a85e2c5bd16c353e445ad8b691bd3868a69954/asyncua/server/server.py#L81

**Reply by Thomangin** [reply]
Thank you for the quick answer, works like a charm.
